### PR TITLE
feat(docs): No generic "Expo app" usage

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Help convey the idea that all tools provided by Expo work with any React Native app by stripping generic usage of "Expo" as in "Expo app".

Left a couple usages in "how expo works" since it needs to be completely rewritten anyways.

### CODE REVIEW

File: docs/pages/push-notifications/using-fcm.md [hunk 1]
```diff
@@ -13,38 +13,38 @@ Note that FCM is not currently available for Expo iOS apps.
...
+          "current_key" "<your Google Cloud Platform API key>",
+        }
+      ],
+    }
+  ]
+}
+```
 
-  > **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**
+> **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**
```
Comment (Simek) [inline docs/pages/push-notifications/using-fcm.md:47]: Those indents ensure that content is properly nested under OL entry.

Screenshot 2022-09-01 113504

**Review by Simek — APPROVED** [review]
I really like those changes and the idea behind them, thanks! 👍 

Just please remember to revert the marked whitespace changes before the merge.

File: docs/pages/get-started/errors.md [hunk 2]
```diff
@@ -10,15 +10,15 @@ A Redbox error is displayed when a fatal error prevents your app from running. A
...
+> This is a brief introduction to debugging an React Native app with Expo CLI. For in-depth information, see [Debugging](/workflow/debugging) documentation.
 
 ## Redbox errors and stack traces
 
 When you encounter an error during development, you'll see the error message and a **stack trace**, which is a report of the recent calls your application made when it crashed. This stack trace is shown both in your terminal and the Expo Go app.
 
 This stack trace is **extremely valuable** since it gives you the location of the error's occurrence. For example, in the following image, the error comes from the file **HomeScreen.js** and is caused on line 7 in that file.
 
-<ImageSpotlight alt="An example of Stack trace in an Expo app." src="/static/images/stack-trace.jpg" style={{maxWidth: 720}} />
+<ImageSpotlight alt="An example of Stack trace in a React Native app." src="/static/images/stack-trace.jpg" style={{maxWidth: 720}} />
```
Comment (kbrandwijk) [inline docs/pages/get-started/errors.md:21]:
```suggestion
<ImageSpotlight alt="An example of a Stack trace in a React Native app." src="/static/images/stack-trace.jpg" style={{maxWidth: 720}} />
```

Not related to your changes, but it caught my eye.

File: docs/pages/guides/authentication.md [hunk 3]
```diff
@@ -2314,7 +2314,7 @@ https://auth.expo.io/@johnsample/my-app
 
 - **Environment:** Development or production projects in Expo Go, or in a standalone build.
 - **Create:** Use `AuthSession.makeRedirectUri({ useProxy: true })` to create this URI.
-  - The link is constructed from your Expo username and the Expo app name, which are appended to the proxy website.
+  - The link is constructed from your Expo username and the app's EAS Update `slug`, which are appended to the proxy website.
```
Comment (kbrandwijk) [inline docs/pages/guides/authentication.md:2317]: 'slug' isn't something specific to EAS Update, in fact the opposite.

File: docs/pages/guides/authentication.md [hunk 4]
```diff
@@ -2330,7 +2330,7 @@ exp://u.expo.dev/F767ADF57-B487-4D8F-9522-85549C39F43F?channel-name=main&runtime
 
 - **Environment:** Production projects that you `eas update`'d and opened in the Expo Go app.
 - **Create:** Use `AuthSession.makeRedirectUri({ useProxy: false })` to create this URI.
-  - The link is constructed from your Expo username and the Expo app name, which are appended to the Expo Go app URI scheme.
+  - The link is constructed from your Expo username and the app's EAS Update `slug`, which are appended to the Expo Go app URI scheme.
```
Comment (kbrandwijk) [inline docs/pages/guides/authentication.md:2333]: 'slug' isn't specific to EAS Update, in fact the opposite.

File: docs/pages/guides/environment-variables.md [hunk 5]
```diff
@@ -25,28 +27,24 @@ export default {
...
-To use these `.extra` properties in your Expo app, you have to use the [`expo-constants`](../versions/latest/sdk/constants.md) module. Here you can see a simple component rendering the comments component, only when these are enabled.
+To use these `extra` properties in your app, you have to use the [`expo-constants`](/versions/latest/sdk/constants.md) module. Here you can see a simple component rendering the comments component, only when these are enabled.
 
-```js
+```tsx
 import Constants from 'expo-constants';
 
-export const Post = props => (
-  <View>
+export const Post = ({ enableComments = Constants.manifest.extra.enableComments || false }) => (
```
Comment (kbrandwijk) [inline docs/pages/guides/environment-variables.md:35]: Isn't `Constants.expoConfig` the recommended way to access this now?

File: docs/pages/guides/authentication.md (hunk 3 above)
Comment (EvanBacon) [reply docs/pages/guides/authentication.md:2317]:
```suggestion
  - The link is constructed from your Expo username and the Expo config (**app.json**) `slug`, which are appended to the proxy website.
```

File: docs/pages/guides/authentication.md (hunk 4 above)
Comment (EvanBacon) [reply docs/pages/guides/authentication.md:2333]:
```suggestion
  - The link is constructed from your Expo username and the Expo config (**app.json**) `slug`, which are appended to the Expo Go app URI scheme.
```

File: docs/pages/guides/environment-variables.md (hunk 5 above)
Comment (amandeepmittal) [reply docs/pages/guides/environment-variables.md:35]: @jonsamp opened a PR to update those changes: https://github.com/expo/expo/pull/18956#pullrequestreview-1095477892

**Review by amandeepmittal — APPROVED** [review]
Overall, these changes look great! 

Thank you for doing this! Just a left comment about updating to `Constants.expoCofig` which Jon also has a PR opened for.

File: docs/pages/guides/environment-variables.md (hunk 5 above)
Comment (kbrandwijk) [reply docs/pages/guides/environment-variables.md:35]: Okay, so let's do it in this PR right away, because his PR doesn't contain this change yet.

**Pull request merged**